### PR TITLE
[core] Only enable D-Cache without `:platform:dma` module

### PR DESCRIPTION
**Pull request by salkinium**
A follow up to https://github.com/modm-io/modm/pull/1222#issuecomment-2468886394.
Relates to #485.

### CODE REVIEW

File: src/modm/platform/core/cortex/module.lb [hunk 1]
```diff
@@ -330,17 +330,25 @@ def validate(env):
 
 def build(env):
     env.substitutions = env.query("vector_table")
+    core = env.substitutions["core"]
+    with_icache = "m7" in core
+    with_dcache = with_icache and not env.has_module(":platform:dma")
```
Comment (chris-durand) [inline src/modm/platform/core/cortex/module.lb]: The H7 BDMA driver is in a separate module, `:platform:bdma`

File: src/modm/platform/core/cortex/module.lb [hunk 2]
```diff
@@ -330,17 +330,25 @@ def validate(env):
...
-        "with_msplim": sum(c.isnumeric() for c in env.substitutions["core"]) == 2,
+        "with_msplim": sum(c.isnumeric() for c in core) == 2,
+        "with_icache": with_icache,
+        "with_dcache": with_dcache,
     })
     env.outbasepath = "modm/src/modm/platform/core"
 
+    if env.substitutions["with_icache"] and not env.substitutions["with_dcache"]:
+        env.warning("Cortex-M7 D-Cache is disabled due to using DMA!")
+
```
Comment (chris-durand) [inline src/modm/platform/core/cortex/module.lb]:
```
Generating: samv71_xplained_ultra/dma/block_transfer
ERROR: In 'modm:platform:cortex-m':
Traceback (most recent call last):
  File "/__w/modm/modm/src/modm/platform/core/cortex/module.lb", line 350, in build
    env.warning("Cortex-M7 D-Cache is disabled due to using DMA!")
AttributeError: 'EnvironmentBuildFacade' object has no attribute 'warning'
```

File: src/modm/platform/core/cortex/startup.c.in [hunk 3]
```diff
@@ -96,26 +98,28 @@ void __modm_startup(void)
...
-
-%% if "m7" in core
+%#
+%% if with_icache
 	// Enable instruction cache
 	SCB_InvalidateICache();
 	SCB_EnableICache();
+%% endif
+%% if with_dcache
 	// Enable data cache with write-through policy
```
Comment (chris-durand) [inline src/modm/platform/core/cortex/startup.c.in]: wrong comment about write-through

File: src/modm/platform/core/cortex/module.lb (hunk 2 above)
Comment (salkinium) [reply src/modm/platform/core/cortex/module.lb]: 🙈

File: src/modm/platform/core/cortex/startup.c.in (hunk 3 above)
Comment (salkinium) [reply src/modm/platform/core/cortex/startup.c.in]: Mondays…

**Review by chris-durand — APPROVED** [review]

**Review by chris-durand — APPROVED** [review]
Thanks!

**Pull request merged**